eng/Versions.props

```
@@ -122,6 +122,10 @@
     <!-- Dependencies from https://github.com/NuGet/NuGet.Client -->
     <NuGetBuildTasksPackageVersion>6.8.1-rc.2</NuGetBuildTasksPackageVersion>
   </PropertyGroup>
+  <PropertyGroup>
+    <!-- Dependencies from https://github.com/dotnet/msbuild -->
+    <MicrosoftBuildPackageVersion>17.8.5</MicrosoftBuildPackageVersion>
```

```suggestion
    <MicrosoftBuildPackageVersion>17.10.6</MicrosoftBuildPackageVersion>
```

@marcpopMSFT, review on the version number?

---

(I looked at the first error, and it seemed to be that it couldn't find 17.8.5, only 17.9.5, but I believe 8.0.3xx should be 17.10.*, so hopefully it has this version, too?